splitter/processor.go

```diff
@@ -39,55 +39,75 @@ func createMessage(index int, row []string) Message {
 
 func (p *Processor) Process(r io.Reader) {
 
-		csvR := csv.NewReader(r)
-		var index = 0
-		var errorCount = 0
+	csvR := csv.NewReader(r)
+	var index = 0
+	var batchSize = config.BatchSize
+	var batchNumber = 1
+	var isFinalBatch = false
+
+batchLoop:
+	for { // each batch
+
+		log.Debug("Processing batch number "+strconv.Itoa(batchNumber)+" index: "+strconv.Itoa(index), nil)
+		var msgs []*sarama.ProducerMessage = make([]*sarama.ProducerMessage, batchSize)
+
+	createBatchLoop:
+		for batchIndex := 0; batchIndex < batchSize; batchIndex++ { // each row in the batch
 
-		csvLoop:
-		for {
 			row, err := csvR.Read()
 			if err != nil {
 				if err == io.EOF {
 					log.Debug("EOF reached, no more records to process", nil)
-					break csvLoop
+					isFinalBatch = true
+					msgs = msgs[0:batchIndex] // the last batch is smaller than batch size, so resize the slice.
+					log.Debug(strconv.Itoa(batchIndex)+" messages in the final batch.", nil)
+					break createBatchLoop
 				} else {
 					fmt.Println("Error occored and cannot process anymore entry", err.Error())
 					panic(err)
```

Comment:
i know its not part of the PR, but a panic seems extreme since it'll kill the entire app